src/lib/parse/contracts.js

```diff
@@ -231,116 +243,108 @@ const getSchema = (body: BodyDescriptor): ?JsonSchema => {
         if (!contractAction) {
-            logger.error(`${action.method} ${resource.uriTemplate} is not in the contract`);
+            logger.error(`${fixtureAction.method} ${resource.uriTemplate} is not in the contract`);
+            return;
+        }
+        if (!fixtureAction.examples || !fixtureAction.examples.length) {
+            logger.error(`${fixtureAction.method} ${resource.uriTemplate} has no examples in the contract`);
             return;
         }
-        if (!action.examples) return;
         const validExamples: Array<Example> = [];
 
-        action.examples.forEach((example, exampleIndex) => {
-            const validRequests: Array<BodyDescriptor> = [];
-            const validResponses: Array<BodyDescriptor> = [];
+        fixtureAction.examples.forEach((example, exampleIndex) => {
 
-            //expect requests and responses in pairs - otherwise it is hard to reason about author's expectations
+            //enforce requests and responses in pairs - otherwise it is hard to reason about author's expectations
             if (example.requests.length > 1 ||  example.responses.length > 1) {
                 throw new Error(`Found more than one request or response for example ${exampleIndex}. Requests and responses expected in pairs.`);
             }
+            const request = example.requests[0];
+            const response = example.responses[0];
 
-            if (action.method === 'POST' || action.method === 'PUT') {
+            const requestValid = validateBody(request, contractAction.request);
 
-                for (let requestIndex = 0; requestIndex < example.requests.length; requestIndex++) {
-                    const metadata: BodyMetadata = { method: action.method, url: resource.uriTemplate, exampleIndex, bodyIndex: requestIndex };
-
-                    const request = example.requests[requestIndex];
-                    const requestResult = validateBody(request, contractAction.request, 'request', metadata);
-
-                    const response = example.responses[requestIndex];
-                    const responseResult = validateBody(response, contractAction.response, 'response', metadata);
+            if (!requestValid.valid) {
+                logger.error(`${fixtureAction.method} ${resource.uriTemplate} example[${exampleIndex}] request ${requestValid.message}`);
+                // if request isn't valid we don't care if responses are
+                return;
+            }
 
-                    if (requestResult.valid && responseResult.valid) {
-                        validRequests.push(request);
-                        validResponses.push(response);
-                    }
+            let responseValid = false;
+            const errors = [];
+            for (let i = 0; i < contractAction.responses.length; i++) {
+                const bodyValidationResult = validateResponse(response, contractAction.responses[i]);
```

Comment:
consider inlining validateResponse to pull validateBody up into the caller, so we can see the same method being used for requests and responses.